web3/manager.py

```diff
@@ -26,13 +37,20 @@
 )
 from web3.providers import (
     AutoProvider,
+    BaseProvider,
+)
+from web3.types import (
+    JsonRpcResponse,
+    Middleware,
 )
 
+if TYPE_CHECKING:
+    from web3 import Web3  # noqa: F401
 
-def apply_error_formatters(
-        error_formatters,
-        response):
 
+def apply_error_formatters(
+    error_formatters: Dict[str, Any], response: JsonRpcResponse
```

Comment:
type looks wrong for `error_formatters`.  I'd expect it to be a callable since it's being used in `pipe`.